import-scripts/update-az-mskimpact.sh

```diff
@@ -0,0 +1,141 @@
+# 3. Remove Part C non-consented patients + samples
+printTimeStampedDataProcessingStepMessage "subset and merge of MSK-IMPACT Part C Consented patients for AstraZeneca"
+
+# Generate subset of Part C consented patients from MSK-Impact
+$PYTHON_BINARY $PORTAL_HOME/scripts/generate-clinical-subset.py \
+    --study-id="mskimpact" \
+    --clinical-file="$AZ_TMPDIR/data_clinical_patient.txt" \
+    --filter-criteria="PARTC_CONSENTED_12_245=YES" \
+    --subset-filename="$AZ_TMPDIR/az_msk_impact_subset.txt"
+
+if [ $? -gt 0 ] ; then
+    echo "ERROR! Failed to generate subset of MSK-IMPACT for AstraZeneca. Skipping merge and update of AstraZeneca MSK-IMPACT!"
+    sendPreImportFailureMessageMskPipelineLogsSlack "AstraZeneca subset generation from MSK-IMPACT"
+
+    EMAIL_BODY="Failed to subset AstraZeneca MSK-IMPACT data. Subset study will not be updated."
+    echo -e "Sending email $EMAIL_BODY"
+    echo -e "$EMAIL_BODY" | mail -s "AstraZeneca MSK-IMPACT Subset Failure: Study will not be updated." $PIPELINES_EMAIL_LIST
+
+    cd $AZ_DATA_HOME ; $GIT_BINARY reset HEAD --hard
+
+    exit 1
+fi
+
+# Write out the subsetted data
+$PYTHON_BINARY $PORTAL_HOME/scripts/merge.py \
+    --study-id="mskimpact" \
+    --subset="$AZ_TMPDIR/az_msk_impact_subset.txt" \
+    --output-directory="$AZ_MSK_IMPACT_DATA_HOME" \
+    --merge-clinical="true" \
+    $AZ_TMPDIR
+
+if [ $? -gt 0 ] ; then
+    echo "Error! Failed to merge subset of MSK-IMPACT for AstraZeneca. Skipping update of AstraZeneca MSK-IMPACT!"
+    sendPreImportFailureMessageMskPipelineLogsSlack "AstraZeneca subset merge from MSK-IMPACT"
+
+    EMAIL_BODY="Failed to merge subset of MSK-IMPACT for AstraZeneca"
+    echo -e "Sending email $EMAIL_BODY"
+    echo -e "$EMAIL_BODY" |  mail -s "AstraZeneca MSK-IMPACT Merge Failure: Study will not be updated." $PIPELINES_EMAIL_LIST
+
+    cd $AZ_DATA_HOME ; $GIT_BINARY reset HEAD --hard
+
+    exit 1
+fi
+
+# Remove temporary directory now that the subset has been merged
+rm -rf "$AZ_TMPDIR"
+
+# ------------------------------------------------------------------------------------------------------------------------
+# 4. Run changelog script
+printTimeStampedDataProcessingStepMessage "generate changelog for AstraZeneca MSK-IMPACT updates"
+
+$PYTHON3_BINARY $PORTAL_HOME/scripts/generate_az_study_changelog.py $AZ_MSK_IMPACT_DATA_HOME
+
+if [ $? -gt 0 ] ; then
+    echo "Error! Failed to generate changelog summary for AstraZeneca MSK-Impact subset."
+    sendPreImportFailureMessageMskPipelineLogsSlack "AstraZeneca MSK-IMPACT changelog generation "
+
+    EMAIL_BODY="Failed to generate changelog summary for AstraZeneca MSK-Impact subset"
+    echo -e "Sending email $EMAIL_BODY"
+    echo -e "$EMAIL_BODY" |  mail -s "AstraZeneca MSK-IMPACT Changelog Failure: Changelog summary will not be provided." $PIPELINES_EMAIL_LIST
+
+    cd $AZ_DATA_HOME ; $GIT_BINARY reset HEAD --hard
+
+    exit 1
```

Comment:
we could ask AZ : do you want a data update if the change report failed to be produced that day? Or should the data update be delayed until we fix whatever caused the failure to create the changelog [possibly a data problem]?